src/decirest_collection_handler.erl

```diff
@@ -27,7 +27,8 @@ replace_and_add/3,
 ]).
 
 -spec init(_,map()) -> {'cowboy_rest',_,#{'rstate':=#{}, _=>_}}.
-init(Req, State) ->
+init(Req, State0) ->
+  State = decirest_router:maybe_reroute_to_module(Req, State0),
```

Comment:
isn't it better to add a init callback here, and if you choosen to implement a version system or anything else you do a reroute, or model change.

I can thing of a few loose thing where this is desirable. If so it shouldn't be maybe but `decirest:change_module(CurrentModule, NewModule)` or something. The maybe should be implemented in the system using decirest